doozerlib/cli/release_gen_payload.py

```diff
@@ -0,0 +1,334 @@
+                if not embargoed:  # exclude embargoed images from the ocp[-arch] imagestreams
+                    runtime.logger.info(f"Adding {arch} image {pullspecs[-1]} to the public mirroring list with imagestream tag {tag_name}...")
+                    mirroring.setdefault(arch, {})[tag_name] = mirroring_value
+                else:
+                    red_print(f"Found embargoed image {pullspecs[-1]}")
+                if runtime.group_config.public_upstreams:
+                    # when public_upstreams are configured, both embargoed and non-embargoed images should be included in the ocp[-arch]-priv imagestreams
+                    runtime.logger.info(f"Adding {arch} image {pullspecs[-1]} to the private mirroring list with imagestream tag {tag_name}...")
+                    mirroring.setdefault(f"{arch}-priv", {})[tag_name] = mirroring_value
+        if not error:
+            state.record_image_success(lstate, image)
+        else:
+            red_print(error, file=sys.stderr)
+            state.record_image_fail(lstate, image, error, runtime.logger)
+
+    for key in mirroring:
+        private = key.endswith("-priv")
+        arch = key[:-5] if private else key  # strip `-priv` suffix
+
+        mirror_filename = 'src_dest.{}'.format(key)
+        imagestream_filename = 'image_stream.{}'.format(key)
+        target_is_name = is_name
+        target_is_namespace = is_namespace
+        if arch != 'x86_64':
+            target_is_name = '{}-{}'.format(target_is_name, arch)
+            target_is_namespace = '{}-{}'.format(target_is_namespace, arch)
+
+        if private:
+            target_is_namespace += "-priv"  # should we also append `-priv` suffix to target_is_name?
+
+        def build_dest_name(tag_name):
+            entry = mirroring[key][tag_name]
+            if arch == 'x86_64':
+                arch_ext = ''
+            else:
+                arch_ext = '-{}'.format(arch)
+            return 'quay.io/{orgrepo}:{version}-{release}{arch_ext}-ose-{tag_name}-{uuid}'.format(orgrepo=orgrepo,
+                                                                                                  version=entry['version'],
+                                                                                                  release=entry['release'],
+                                                                                                  arch_ext=arch_ext,
+                                                                                                  tag_name=tag_name,
+                                                                                                  uuid=runtime.uuid)
+
+        # Save the default SRC=DEST 'oc image mirror' input to a file for
+        # later.
+        with io.open(mirror_filename, 'w+', encoding="utf-8") as out_file:
+            for tag_name in mirroring[key]:
+                dest = build_dest_name(tag_name)
```

Comment:
I'd like the tags for our destinations to no longer include any identifying information - no dates, no runtime.uuid. I do not want an attacker to be able to derive a tag for an embargoed image based on a tag name for an unembargoed image. The easiest way I know how to do this to tag the image with a string including its own sha.
So instead of `quay.io/{orgrepo}:{version}-{release}{arch_ext}-ose-{tag_name}-{uuid}` , something like `quay.io/{orgrepo}:sha-{image_sha}`